[feat/aut819] Implement version bumper

ChristopherAbram opened the pull request:
+ Provide script for reading and updating smartpeak version from CMake configuration (smartpeak tool python package)
+ Provide scripts for bumping/setting smartpeak version from commit message
+ Update smartpeak version to the latest released

Note: I placed the package in the `tools` subdirectory.

ChristopherAbram committed:
[feat/aut819] Implement version bumper
+ Provide script for reading and updating smartpeak version from CMake configuration (smartpeak tool python package)
+ Provide scripts for bumping/setting smartpeak version from commit message
+ Update smartpeak version to the latest released

bertrandboudaud commented on tools/smartpeak/smartpeak_git_prepare_release:
```diff
@@ -0,0 +1,32 @@
+        success, (n_major, n_minor, n_patch) = utils.get_new_version(
+            set_version, bump, (major, minor, patch))
+        if not success:
+            return 1
```
We don't need any error message?

ChristopherAbram replied:
There are error messages in inside methods, just return code has to go be according in main script

bertrandboudaud commented on tools/smartpeak/smartpeak_git_prepare_release:
```diff
@@ -0,0 +1,32 @@
+        # Update config:
+        success = utils.update_version(config_file, version=(n_major, n_minor, n_patch))
+        if not success:
+            return 1
```
We don't need any error message?

bertrandboudaud reviewed (approved):
looks great. ok for me

ChristopherAbram commented on tools/smartpeak/smartpeak_prepare_release:
```diff
@@ -0,0 +1,44 @@
+
+import smartpeak.utils as utils
+
+
```
Good point, I will add this while adding developer documentation

ChristopherAbram committed:
Merge branch 'feat/aut819' into develop

dmccloskey reviewed (changes requested):
Looks very nice.

If you can 1) add in docstrings to the methods, 2) provide unit tests for the new methods, and 3) take care of the DeepCode issues, that would be great.

ChristopherAbram committed:
[feat/aut819] Implement version bumper
+ Add script which returns version after bumping (dry mode)
+ Update utils.py file with bumped_version() function
+ Add bash release script,
+ Add documentation and unit tests.

ChristopherAbram committed:
Merge branch 'develop' of https://github.com/AutoFlowResearch/SmartPeak into develop

ChristopherAbram committed:
Merge branch 'develop' of https://github.com/AutoFlowResearch/SmartPeak into develop

dmccloskey reviewed (changes requested):
Docstrings and unit tests look great.  Thanks for adding those in.  The only thing I see left is adding the running of the unit tests into the CircleCI pipelines.  Ping me when it is ready and we can merge.

ChristopherAbram committed:
fix: move python tests to test directory,
ci: add running python tests for the tool package

ChristopherAbram committed:
fix: run python tests

ChristopherAbram committed:
fix: run python tests

ChristopherAbram committed:
fix: set pythonpath

ChristopherAbram committed:
fix: python tests

ChristopherAbram committed:
[feat/aut819] Implement version bumper
- Add `smartpeak_send_emails` to automate sending custom emails.
- Add execute permission to every smartpeak_* command.

dmccloskey commented on tools/smartpeak/smartpeak_send_emails:
```diff
@@ -0,0 +1,158 @@
+
+      msgPlain = '''\
+SmartPeak v{version} is released!\n
+We are constantly working to improve your SmartPeak experience, below a summary of what has changed in the new release.\n
```
Very nice.

dmccloskey commented on tools/smartpeak/smartpeak_send_emails:
```diff
@@ -0,0 +1,158 @@
+SCOPES = ['https://www.googleapis.com/auth/gmail.send']
+
+
+def create_message(sender, to, subject, msgHtml, msgPlain):
```
Did you add in unit tests for these new methods?

ChristopherAbram replied:
There is not really anything to test here, there two helper functions which only wraps message in MIME object and then content is encoded with base64. I would leave it as it is for now.

dmccloskey reviewed (changes requested):
Additions look very nice.  My only minor comment the missing unit tests for the new python methods for the email functionality.

ChristopherAbram committed:
fix: Change links in email content
